src/03-modality-agnostic-files.md

```diff
@@ -396,6 +396,284 @@ any identifiable information with the source code.
 There are no limitations or recommendations on the language and/or
 code organization of these scripts at the moment.
 
+<sup>1</sup>Storing actual source files with the data is preferred over links to
+external source repositories to maximize long term preservation (which would
+suffer if an external repository would not be available anymore).
+
+## Provenance of BIDS datasets, files, and derivatives
+
+Template:
+
+```Text
+- [Dataset level] prov.jsonld
+- [File level] sub-<label>/[ses-<label>/]sub-<label>[_ses-<label>]_<suffix>.prov
```

Comment:
Thinking about many types of "applications" operating on entire subject/sessions, indeed it makes sense to have it at the subj/session level (which is presented here, not file level) and have possibility for file level too to "complement" that (similarly how done in inheritance principle for other metadata?):

```suggestion
- [Subject[/session] level] sub-<label>/[ses-<label>/]sub-<label>[_ses-<label>]_<suffix>.prov
- [File level] sub-<label>/[ses-<label>/]<modality>/sub-<label>[_ses-<label>][_entities-<values>]_<suffix>.prov
```

I am also not sure why it is `prov.jsonld` on top and `.prov` and the lower?  why not `.prov.jsonld` at the bottom levels too?

Reply:
What about pretty much allowing `.jsonld` at any level where appropriate?  E.g. could be for subject session or specific modality processing etc. All of those to be joined into a single graph anyways, right?

Reply:
pushed a fix for `.prov`